src/deepsparse/loggers/metric_functions/helpers/config_generation.py

```diff
@@ -0,0 +1,123 @@
+# Copyright (c) 2021 - present / Neuralmagic, Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Helper functions for generating metric function configs
+"""
+
+__all__ = ["data_logging_config_from_predefined"]
+
+import textwrap
+from typing import Any, Dict, List, Optional, Union
+
+from deepsparse.loggers.config import MetricFunctionConfig
+
+
+_WHITESPACE = "  "
+
+
+def data_logging_config_from_predefined(
+    group_names: Union[str, List[str]],
+    loggers: Dict[str, Optional[Dict[str, Any]]] = {"python": {}},
```

Comment:
been leaving this comment a lot - we cannot use mutable types as function defaults in python as they can have unintended side effects. read more here: https://florimond.dev/en/posts/2018/08/python-mutable-defaults-are-the-source-of-all-evil/

Reply:
(moving discussion to #867 since this will go away after rebase)

Reply:
Interesting!

Reply:
I actually now see that this does mess up my code.

Reply:
I actually do experience it right now but with a different function (inside the same module):
```
def _nested_dict_to_lines(
    value: Any, key=None, yaml_str_lines=[], _level=0
) -> List[str]:```

such a counterintuitive thing....